ArduCopter/GCS_Mavlink.cpp

```diff
@@ -97,15 +97,30 @@ void GCS_MAVLINK_Copter::send_position_target_global_int()
         return;
     }
 
-    // convert altitude frame to AMSL (this may use the terrain database)
-    if (!target.change_alt_frame(Location::AltFrame::ABSOLUTE)) {
-        return;
+    // Note, currently get_wp() only returns either ABOVE_TERRAIN or ABOVE_ORIGIN
+    MAV_FRAME frame;
+    switch (target.get_alt_frame()) {
+    case Location::AltFrame::ABOVE_TERRAIN:
+        frame = MAV_FRAME_GLOBAL_TERRAIN_ALT;
+        break;
+    case Location::AltFrame::ABSOLUTE:
+        frame = MAV_FRAME_GLOBAL;
+        break;
+    case Location::AltFrame::ABOVE_HOME:
+        frame = MAV_FRAME_GLOBAL_RELATIVE_ALT;
+        break;
+    case Location::AltFrame::ABOVE_ORIGIN:
```

Comment:
Hummm this one don't seems right. I will look what it should be

Reply:
Yeah. This isn't 100% as there is no equivalent mavlink_frame for the ekf origin. Really what should eventually be done is to ensure that the calling modes set the altitude frame according to the current commanded waypoint's frame.

Reply:
MAV_FRAME_LOCAL_NED

Reply:
but according to mavlink specs, valid frame are : 
Valid options are: MAV_FRAME_GLOBAL_INT = 5, MAV_FRAME_GLOBAL_RELATIVE_ALT_INT = 6, MAV_FRAME_GLOBAL_TERRAIN_ALT_INT = 11

Reply:
So I would say, if that is in the specs, use the right frame otherwise default to MAV_FRAME_GLOBAL_INT even if that is a change in behavior

Reply:
Yeah. That is why I defaulted EKF_origin to become MAV_FRAME_GLOBAL as is currently being done. From my reading it seems that *_INT part has been dropped since it is redundant.

Reply:
This should be `location_alt_frame_to_mavlink_coordinate_frame`, like we have `bool GCS_MAVLINK::mavlink_coordinate_frame_to_location_alt_frame(const MAV_FRAME coordinate_frame, Location::AltFrame &frame)`

Even if it is a bit of a mouthful.

That won't work if we really do need to change the number, of course.

Reply:
Added `location_alt_frame_to_mavlink_coordinate_frame` to GCS_MAVLINK.

Is the comment of, "That won't work if we really do need to change the number, of course.", about the *_INT of the mavlink frame?